modules/mysql/read_replica.tf

```diff
@@ -25,7 +25,7 @@ resource "google_sql_database_instance" "replicas" {
   for_each             = local.replicas
   project              = var.project_id
   name                 = "${local.master_instance_name}-replica${var.read_replica_name_suffix}${each.value.name}"
-  database_version     = var.database_version
+  database_version     = var.database_version_replica != "" ? var.database_version_replica : var.database_version
```

Comment:
This logic should look at the `read_replicas` slice and try to pull the `database_version` from it. If not found, the `database_version` for the primary should be used.